site/docs/guide/files.md

```diff
@@ -5,107 +5,112 @@ next: ./games.md
-Note that the URL is only valid for at least 60 minutes, then it may expire and you have to call `getFile` again to obtain a new link.
+Whenever your bot **receives** a message with a file, it will in fact not directly receive the complete file data, but only the `file_id` instead.
+If your bot actually wants to download the file, then it can do so by calling the `getFile` method ([Telegram Bot API reference](https://core.telegram.org/bots/api#getfile)).
+This method enables you to download the file by constructing a special, temporary, URL.
+Note that this URL is only guaranteed to be valid for 60 minutes, after which it may expire. In this case, you can simply call `getFile` again.
 
-When a bot **sends** a message, it can specify a `file_id` that is has seen before.
-This will then send the identified file.
-(You can also upload your own files, [scroll down](#sending-files) to see how.)
-You can reuse the same `file_id` as often as you want, so you could send the same file to five different chats, always with the same `file_id`.
-Make sure to use the correct method—for instance, you cannot use a `file_id` that identifies a photo when calling [`sendVideo`](https://core.telegram.org/bots/api#sendvideo).
+When a bot **sends** a message, it can specify a `file_id` that it has seen before.
+This will allow it to send the identified file, without needing to upload the data for it.
+(To see how to upload your own files, [scroll down](#sending-files).)
+You can reuse the same `file_id` as often as you want, so you could send the same file to five different chats, using the same `file_id`.
+However, you must make sure to use the correct method—for example, you cannot use a `file_id` that identifies a photo when calling [`sendVideo`](https://core.telegram.org/bots/api#sendvideo).
 
 Every bot has its own set of `file_id`s for the files that it can access.
-You cannot give the `file_id` to your friend's bot to download it there, as another bot will use a different identifier for the same file.
-This implies that you cannot simply guess a `file_id` and access some random person's file, because Telegram keeps track of what `file_id`s are valid for your bot.
+You cannot reliably use a `file_id` from your friend's bot, to access a file with _your_ bot. Each bot will use different identifiers for the same file.
+This implies that you cannot simply guess a `file_id` and access some random person's file, because Telegram keeps track of which `file_id`s are valid for your bot.
 
-On the other hand, it can happen that the same file is identified by different `file_id`s even for the same bot.
-That means that you cannot reliably compare `file_id`s to check if two files are the same.
-In cases where you do need to identify the same file over time and across bots, you should use the value for `file_unique_id` that your bot receives along with every `file_id`.
-The `file_unique_id` cannot be used to download files.
+::: warning Using Foreign file_ids
```

Comment:
Where is the source of this information? Are you talking about collisions?

Reply:
No, not collisions. Different bots accessing the same file with the same `file_id`. For one of these bots (whichever the `file_id` does not belong to), this will eventually stop working.

This is from personal experience, but I have found an official source for you also:
https://t.me/tdlibchat/12134

Reply:
That's not how I interpret what he said. I don't think he meant to say “sure you can use other bot's file identifiers, but only for a limited and unspecified time.”

I read it as “people think they're using the same identifier while in reality they're not.”

Note that the official docs are clear: https://core.telegram.org/bots/api#sending-files

They say that identifiers are unique per bot. I don't see why they'd write that and then implement shared identifiers. Eventual consistency is also not Telegram's style, so I doubt that they'd somehow delay enforcing the correct access rules by a week.

Reply:
It's because they don't want people to reuse the `file_id`s, so they would rather not acknowledge that it is possible. However, it is 100% possible.

Reply:
I am not going to spread this hypothesis until I have seen either official confirmation by Telegram, or a way to reproduce this.